compiler/src/language_server/code_action.re

```diff
@@ -0,0 +1,111 @@
+open Grain;
+open Compile;
+open Grain_parsing;
+open Grain_utils;
+open Grain_typed;
+open Grain_diagnostics;
+open Sourcetree;
+open Lsp_types;
+
+// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#definitionParams
+module RequestParams = {
+  [@deriving yojson({strict: false})]
+  type code_action_context = {diagnostics: list(Protocol.diagnostic)};
+
+  [@deriving yojson({strict: false})]
+  type t = {
+    [@key "textDocument"]
+    text_document: Protocol.text_document_identifier,
+    range: Protocol.range,
+    context: code_action_context,
+  };
+};
+
+// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#locationLink
+module ResponseResult = {
+  [@deriving yojson]
+  type code_action = {
+    title: string,
+    kind: string,
+    edit: Protocol.workspace_edit,
+  };
+
+  [@deriving yojson]
+  type t = option(list(code_action));
+};
+
+let send_no_result = (~id: Protocol.message_id) => {
+  Protocol.response(~id, `Null);
+};
+
+let explicit_type_annotation = (range, uri, version, type_str) => {
+  ResponseResult.{
+    title: "Explicitly annotate type",
```

Comment:
What do you think of just `Annotate type`? If there's no type annotation, then adding one would just be an annotation.

Reply:
Changed to this